kube/resources/worker.yaml

```diff
@@ -0,0 +1,211 @@
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: LOG_LEVEL
+            - name: KUBE_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+            - name: SUBMITTER_NUM_THREADS
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: SUBMITTER_NUM_THREADS
+            - name: RESOURCE_CPU_REQUEST
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: RESOURCE_CPU_REQUEST
+            - name: RESOURCE_CPU_LIMIT
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: RESOURCE_CPU_LIMIT
+            - name: RESOURCE_MEMORY_REQUEST
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: RESOURCE_MEMORY_REQUEST
+            - name: RESOURCE_MEMORY_LIMIT
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: RESOURCE_MEMORY_LIMIT
+            - name: S3_LOG_BUCKET
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: S3_LOG_BUCKET
+            - name: S3_LOG_BUCKET_REGION
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: S3_LOG_BUCKET_REGION
+            - name: AWS_ACCESS_KEY_ID
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: AWS_ACCESS_KEY_ID
+            - name: AWS_SECRET_ACCESS_KEY
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: AWS_SECRET_ACCESS_KEY
+            - name: S3_MINIO_ENDPOINT
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: S3_MINIO_ENDPOINT
+            - name: S3_PATH_STYLE_ACCESS
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: S3_PATH_STYLE_ACCESS
+            - name: GOOGLE_APPLICATION_CREDENTIALS
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: GOOGLE_APPLICATION_CREDENTIALS
+            - name: GCP_STORAGE_BUCKET
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: GCP_STORAGE_BUCKET
+            - name: INTERNAL_API_HOST
+              valueFrom:
+                configMapKeyRef:
+                  name: airbyte-env
+                  key: INTERNAL_API_HOST
+          ports:
+            - containerPort: 9000 # for heartbeat server
+            - containerPort: 9001 # start temporal worker port pool
```

Comment:
does this mean each worker now has the ability to run up to 30 concurrent jobs? so we can theoretically scale up horizontally as we scale up number of workers? 

is this containerPort configuration our ideal state for this configuration? it seems a little weird to have it written out like this, but this may be a reasonable approach?

Reply:
For syncs it requires 4 ports (stderr/stdout for two pods) each. 

Medium term we should use a single port for all of this communication and have separate channels it communicates over (think grpc bidirectional or zeromq or something).